internal/drivers/ankabuild/driver.go

```diff
@@ -0,0 +1,177 @@
+package ankabuild
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/drone-runners/drone-runner-aws/internal/drivers"
+	"github.com/drone-runners/drone-runner-aws/internal/lehelper"
+	"github.com/drone-runners/drone-runner-aws/types"
+	"github.com/drone/runner-go/logger"
+
+	"github.com/sirupsen/logrus"
+)
+
+type config struct {
+	username    string
+	password    string
+	rootDir     string
+	vmID        string
+	userData    string
+	nodeID      string
+	registryURL string
+	authToken   string
+	tag         string
+	ankaClient  Client
+}
+
+const (
+	startupScriptCondition = 0
+	scriptFailHandler      = 2
+	scriptTimeout          = 200
+	ankaRunning            = "Running"
+)
+
+func New(opts ...Option) (drivers.Driver, error) {
+	c := new(config)
+	for _, opt := range opts {
+		opt(c)
+	}
+	if c.ankaClient == nil {
+		c.ankaClient = NewClient(c.registryURL, c.authToken)
+	}
+	return c, nil
+}
+
+func (c *config) Create(ctx context.Context, opts *types.InstanceCreateOpts) (instance *types.Instance, err error) {
+	startTime := time.Now()
+	uData := base64.StdEncoding.EncodeToString([]byte(lehelper.GenerateUserdata(c.userData, opts)))
+	machineName := fmt.Sprintf(opts.RunnerName+"-"+"-%d", startTime.Unix())
```

Comment:
runner names can have spaces in them, and that may cause issues here

Reply:
<img width="1108" alt="Screenshot 2022-08-08 at 15 03 42" src="https://user-images.githubusercontent.com/83226740/183436680-03e9c199-db95-4076-99fb-03ec45df02ea.png">
Not an issue :)